ONSdigital/eq-questionnaire-runner
# Grand calculated summary inside a repeating section

### What is the context of this PR?
This PR adds support for using a Grand Calculated Summary within a repeating section. Within a repeating section a grand calculated summary is allowed to reference other calculated summaries within the repeating section, or non-repeating calculated summaries from other sections.

The [corresponding validator PR](https://github.com/ONSdigital/eq-questionnaire-validator/pull/215) ensures that a GCS can only reference a repeating CS if both reside in the same repeating section

(Note that the schema is actually valid without the validator changes, as those have added restrictions rather than enabled something new, however this PR is pointing at the validator branch to demo that the validator changes are compatible with these ones)

**For this iteration, a repeating GCS using non-repeating calculated summaries with repeating answers is not supported.**

### How to review
It was decided that the refactor of `return_to` parameters into a class would be done separately in a follow on ticket, so it is not a separate piece of work here but to simplify review, the three main commits are

1. [Add return to list parameters](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1210/commits/5129aa086918e4669d7d2c7c07f23ef1fd7e595c)
2. [Changes required to support grand calculated summaries in repeats](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1210/commits/a52c2176ab2a8b704355e3bc33c8155006f1960b)
3. [Functional tests](https://github.com/ONSdigital/eq-questionnaire-runner/pull/1210/commits/bd89e2bc06d794b0243136b53e8db3d45dfe12ee)

the `return_to` parameters have been extended to include `return_to_list_item_id` and `return_to_list_name` so that when using the change links to route to a non-repeating calculated summary and its answers, the continue button has sufficient information to return to the repeating section at the correct list item. These needed to filter through to answer/block/question/calculated summary contexts and handlers

### How to test functionally
Use the `test_grand_calculated_summary_inside_repeating_section` schema and follow the guidance in the first section. Namely, do not add items to the list, that functionality is in the follow on card. But do check that the change links from the repeating sections for vehicles work as expected, you can change answers in both the repeating and non-repeating calculated summaries and return to the correct location, and that progress updates as expected when updating answers in the first section.

### Checklist
